### DISCUSSION TITLE
data import ?

### DISCUSSION THREAD

**Discussion by ghost** [alan-turing-institute/sktime#1344] [Q&A]
Hi all,
I am trying to use sktime for data analysis of astronomical timeseries.
My data does not comes in any of the formats defined here https://www.sktime.org/en/latest/examples/loading_data.html
Reading the documentation I understand that I need to transform my data in nested dataframe.
I try to implement one of the simple examples for forecasting taken from here https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.naive.NaiveForecaster.html
but I get an error that tells that the data format is not right: "ValueError: Data must be univariate, but found a pd.DataFrame"
Given the error, I failed to understand how sktime works, so could I ask if someone please help me to understand where I am wrong and how to import correctly my data in to sktime ?

This is what I have wrote till now:

```
# a lot of code for defining my data in format of lists: 
# time (in Julian date) and pdc1 (my data in ADU (Analog Digital Unit), not normalized to unity)
# each list has a length equal to NP
import sktime as sk
from sktime.utils.data_processing import from_long_to_nested
from sktime.forecasting.naive import NaiveForecaster
# test data type from the example
from sktime.datasets import load_airline   # import this just to check the datatype used
airline = load_airline()
print('airline data type: ',type(airline))     # the result is:  airline data type:  <class 'pandas.core.series.Series'>
# dataframe creation
# define case_id and dim_id to met the requirements of  from_long_to_nested
case_id=np.arange(NP)
dim_id=np.zeros(NP)
# define a Pandas dataframe with my data and the two additional lists
lc = pd.DataFrame(np.column_stack([case_id,dim_id,times,pdc1]),columns=['case_id','dim_id','time','pdc1'])
# convert lc dataframe to a nested one
lc_nested = from_long_to_nested(lc,instance_column_name='case_id',dimension_column_name='dim_id',time_column_name='time',value_column_name='pdc1')
print('lc_nested data type:',type(lc_nested)) # the result is:    lc_nested data type: <class 'pandas.core.frame.DataFrame'>
# forecasting
forecaster = NaiveForecaster(strategy="drift")
# the following line generate the error "ValueError: Data must be univariate, but found a pd.DataFrame"
forecaster.fit(lc_nested) 
# the following line is never reached
y_pred = forecaster.predict(fh=[1,2,3])
```

**Comment by fkiraly** [comment]
@barbierimauro, multiple answers to this:

* generally, it's no longer necessary to convert to nested data frame format - that notebook is outdated and should be replaced by a more recent one, since we now support other common formats as well. It's still work in progress, see below.
* not all forecasters support multivariate forecasting - you can use `all_estimators` to search forecasters by the `scitype:y` tag, which can be `"univariate"`, `"multivariate"`, or `"both"`. I think `NaiveForecaster` currently doesn't, but probably should, on mid-term.
* you can use `ColumnEnsembleForeecaster` from `forecasting.compose` to apply univariate forecasters per column.
* it would be important to understand though what kind of data you have - you mention `case_id`, so do you have a *panel* (multiple independent instances of time series), or a multivariate series (one series, but multiple variables)
* what also could help you is the notebook from pull request #1232, see here:
https://github.com/alan-turing-institute/sktime/blob/datatypes-doc/examples/AA_datatypes_and_datasets.ipynb
this is not merged yet and under review, but the notebook has explanations of the datatypes currently in use, and some of the functionality it explains is already on the main branch. Have a look, would appreciate feedback.

**Reply by fkiraly** [reply]
FYI, @barbierimauro, the PR is now merged - does this help?

**Comment by fkiraly** [comment]
PS: I assume you are using an older version, since `pd.DataFrame` should be ok for univariate forecasters as long as it has only one column.

**Reply by fkiraly** [reply]
Given your post, I was also thinking: if you have a univariate forecaster, should it automatically be applied to all variables independently?
See https://github.com/alan-turing-institute/sktime/issues/1345
